From 88cdec4c0bc40f4e98105a0b69c968fdd94ac7f4 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Sun, 30 Jun 2024 23:17:00 +0200
Subject: [PATCH 1/3] Fix/simplify BUILD_STANDALONE_APP decisionmaking

BUILD_STANDALONE_APP=ON guarantees Qt::Widgets so no need for this check.

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 79f07df..cdd99e1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -10,7 +10,7 @@ if (TARGET Qt::Quick)
     add_subdirectory(map-quick)
     add_subdirectory(map-publictransport-integration)
 endif()
-if ((NOT ANDROID AND TARGET Qt::Widgets) OR BUILD_STANDALONE_APP)
+if (BUILD_STANDALONE_APP)
     add_subdirectory(app)
 endif()
 if (NOT CROSS_COMPILING)
-- 
2.45.2


From 09d0bda83e6de957ad8883926d8b9f9b3adfe762 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Sun, 30 Jun 2024 23:23:22 +0200
Subject: [PATCH 2/3] Use more QT_MIN_VERSION

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bbf417f..6da25de 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,7 +54,7 @@ if (NOT BUILD_TOOLS_ONLY)
 endif()
 
 if (BUILD_TESTING)
-    find_package(Qt6 6.5 REQUIRED COMPONENTS Test Widgets)
+    find_package(Qt6 ${QT_MIN_VERSION} REQUIRED COMPONENTS Test Widgets)
 endif()
 
 if (BUILD_STANDALONE_APP)
-- 
2.45.2


From 74c3516e698a4fc53e45585a5846a170a8f25e2f Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <asturm@gentoo.org>
Date: Sun, 30 Jun 2024 23:25:46 +0200
Subject: [PATCH 3/3] Add missing KF6I18n dependency for BUILD_STANDALONE_APP

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6da25de..dd55077 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,7 +61,7 @@ if (BUILD_STANDALONE_APP)
     if (NOT ANDROID)
         find_package(Qt6 ${QT_MIN_VERSION} REQUIRED COMPONENTS Widgets)
     else()
-        find_package(KF6 REQUIRED COMPONENTS Kirigami2)
+        find_package(KF6 6.3 REQUIRED COMPONENTS I18n Kirigami)
         find_package(Qt6 ${QT_MIN_VERSION} REQUIRED COMPONENTS QuickControls2 Svg)
         find_package(OpenSSL REQUIRED)
         if (NOT DEFINED BREEZEICONS_DIR AND EXISTS ${CMAKE_SOURCE_DIR}/../breeze-icons)
-- 
2.45.2

